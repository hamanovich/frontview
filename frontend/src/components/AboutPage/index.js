import React, { Fragment } from 'react';
import { Switch, Route, Redirect } from 'react-router-dom';
import { Helmet } from 'react-helmet';

import About from './About';
import Faq from './Faq';
import Pricing from './Pricing';
import LatestNews from './LatestNews';
import UsefulLinks from './UsefulLinks';

const AboutPage = () => (
  <Fragment>
    <Helmet>
      <title>Frontview: About</title>
    </Helmet>
    <Switch>
      <Route exact path="/about" component={About} />
      <Route exact path="/about/news" component={LatestNews} />
      <Route exact path="/about/faq" component={Faq} />
      <Route exact path="/about/pricing" component={Pricing} />
      <Route exact path="/about/links" component={UsefulLinks} />
      <Redirect to="/" />
    </Switch>
  </Fragment>
);

export default AboutPage;
