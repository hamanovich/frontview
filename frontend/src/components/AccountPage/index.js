import React from 'react';
import { shape, bool } from 'prop-types';
import { Switch, Redirect } from 'react-router-dom';
import { Helmet } from 'react-helmet';

import Row from 'react-bootstrap/lib/Row';
import Col from 'react-bootstrap/lib/Col';

import Account from './Account';
import AccountEdit from './AccountEdit';
import AccountBar from './AccountBar';
import QListPage from '../QList/QListPage';
import QLists from '../QList/QLists';

import { PropsRoute } from '../../utils/helpers';

import { UserType } from '../../propTypes';

const AccountPage = ({ auth }) => (
  <Row>
    <Helmet>
      <title>Frontview: {auth.user.username}</title>
    </Helmet>
    <Col md={3} sm={4}>
      <AccountBar auth={auth} />
    </Col>
    <Col md={9} sm={8}>
      <Switch>
        <PropsRoute exact path="/me" component={Account} user={auth.user} />
        <PropsRoute exact path="/me/edit" component={AccountEdit} user={auth.user} />
        <PropsRoute exact path="/me/qlist/create" component={QListPage} userId={auth.user._id} />
        <PropsRoute exact path="/me/qlists" component={QLists} userId={auth.user._id} />
        <Redirect to="/" />
      </Switch>
    </Col>
  </Row>
);

AccountPage.propTypes = {
  auth: shape({
    isAuthenticated: bool.isRequired,
    user: UserType.isRequired,
  }).isRequired,
};

export default AccountPage;
