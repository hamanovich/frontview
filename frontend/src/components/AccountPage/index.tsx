import React, { FC } from 'react';
import { Switch, Redirect } from 'react-router-dom';
import { Helmet } from 'react-helmet';

import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';
import Container from 'react-bootstrap/Container';

import Account from './Account';
import AccountEdit from './AccountEdit';
import AccountBar from './AccountBar';
import QListPage from '../QList/QListPage';
import QLists from '../QList/QLists';

import { PropsRoute } from '../../utils/helpers';
import { AccountPageProps } from './models';

const AccountPage: FC<AccountPageProps> = ({ auth }) => (
  <Container>
    <Helmet>
      <title>Frontview: Profile</title>
    </Helmet>
    <Row>
      <Col xl={9} md={{ span: 8, order: 1 }} className="mb-4">
        <Switch>
          <PropsRoute exact path="/me" component={Account} user={auth.user} />
          <PropsRoute
            exact
            path="/me/edit"
            component={AccountEdit}
            user={auth.user}
          />
          <PropsRoute
            exact
            path="/me/qlist/create"
            component={QListPage}
            userId={auth.user._id}
          />
          <PropsRoute
            exact
            path="/me/qlists"
            component={QLists}
            username={auth.user.username}
          />
          <Redirect to="/" />
        </Switch>
      </Col>
      <Col xl={3} md={4}>
        <AccountBar auth={auth} />
      </Col>
    </Row>
  </Container>
);

export default AccountPage;
