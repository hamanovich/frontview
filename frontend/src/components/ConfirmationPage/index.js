import React, { Fragment } from 'react';
import { bool } from 'prop-types';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import jwtDecode from 'jwt-decode';

import compose from 'recompose/compose';
import pure from 'recompose/pure';
import withState from 'recompose/withState';
import lifecycle from 'recompose/lifecycle';

import Jumbotron from 'react-bootstrap/Jumbotron';
import Button from 'react-bootstrap/Button';
import Container from 'react-bootstrap/Container';

import { confirm } from '../../actions/auth';
import { getUser } from '../../actions/signup';
import { addFlashMessage } from '../../actions/flash';

const enhance = compose(
  connect(
    null,
    {
      confirm,
      getUser,
      addFlashMessage,
    },
  ),

  withState('success', 'setSuccess', false),

  lifecycle({
    componentDidMount() {
      const { confirm, getUser, addFlashMessage, match, setSuccess } = this.props;

      confirm(match.params.token)
        .then(() => getUser(jwtDecode(window.localStorage.jwtToken).username))
        .then(() => setSuccess(true))
        .catch(err =>
          addFlashMessage({
            type: 'error',
            text: err.response.data.error,
          }),
        );
    },
  }),

  pure,
);

const Confirmation = ({ success }) => (
  <Jumbotron>
    <Container>
      <h1>Hey!</h1>
      <p className="lead">
        This is a simple hero unit, a simple Jumbotron-style component for calling extra attention
        to featured content or information.
      </p>
      {success && (
        <Fragment>
          <hr />
          <p className="lead">
            <Link to="/">
              <Button variant="success" size="lg">
                Welcome on board!
              </Button>
            </Link>
          </p>
        </Fragment>
      )}
    </Container>
  </Jumbotron>
);

Confirmation.propTypes = {
  success: bool.isRequired,
};

export default enhance(Confirmation);
