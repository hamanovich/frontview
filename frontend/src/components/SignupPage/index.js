import React from 'react';
import { func } from 'prop-types';
import { connect } from 'react-redux';
import { Helmet } from 'react-helmet';

import Row from 'react-bootstrap/lib/Row';
import Col from 'react-bootstrap/lib/Col';
import PageHeader from 'react-bootstrap/lib/PageHeader';
import FontAwesome from 'react-fontawesome';

import { signup, isUserExists } from '../../actions/signup';
import { addFlashMessage } from '../../actions/flash';

import SignupForm from './SignupForm';

export const SignupPage = ({ signup, addFlashMessage, isUserExists }) => (
  <Row>
    <Helmet>
      <title>Frontview: Register. Welcome</title>
    </Helmet>
    <Col md={6} mdOffset={3}>
      <PageHeader>
        <FontAwesome name="user-plus" /> Register. Welcome
      </PageHeader>

      <SignupForm signup={signup} addFlashMessage={addFlashMessage} isUserExists={isUserExists} />
    </Col>
  </Row>
);

SignupPage.propTypes = {
  signup: func.isRequired,
  addFlashMessage: func.isRequired,
  isUserExists: func.isRequired,
};

export default connect(
  null,
  {
    signup,
    addFlashMessage,
    isUserExists,
  },
)(SignupPage);
