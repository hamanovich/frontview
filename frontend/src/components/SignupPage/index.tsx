import React, { FC } from 'react';
import { connect } from 'react-redux';
import { Helmet } from 'react-helmet';

import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';
import Container from 'react-bootstrap/Container';
import FontAwesome from 'react-fontawesome';

import { signup, isUserExists } from '../../actions/signup';
import { addFlashMessage } from '../../actions/flash';

import SignupForm from './SignupForm';
import { SignupProps } from './models';

export const SignupPage: FC<SignupProps> = ({ signup, addFlashMessage, isUserExists }) => (
  <Container>
    <Helmet>
      <title>Frontview: Register. Welcome</title>
    </Helmet>
    <Row>
      <Col xl={{ span: 6, offset: 3 }} md={{ span: 8, offset: 2 }}>
        <h1>
          <FontAwesome name="user-plus" /> Register. Welcome
        </h1>

        <SignupForm signup={signup} addFlashMessage={addFlashMessage} isUserExists={isUserExists} />
      </Col>
    </Row>
  </Container>
);

export default connect(
  null,
  {
    signup,
    addFlashMessage,
    isUserExists,
  },
)(SignupPage);
