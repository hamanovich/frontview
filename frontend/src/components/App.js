import React, { Fragment, lazy } from 'react';
import { Switch, Route } from 'react-router-dom';
import { Helmet } from 'react-helmet';

import HomePage from './HomePage';
import NotFound from './NotFound';
import Header from './layout/Header';
import Footer from './layout/Footer';
import FlashList from './flash/FlashList';
import FaqButton from './shared/FaqButton';

import { isLoggedIn, User, Waiting } from '../utils/helpers';

const LoginPage = lazy(() => import('./LoginPage' /* webpackChunkName: "Login" */));
const SignupPage = lazy(() => import('./SignupPage' /* webpackChunkName: "Signup" */));
const AccountPage = lazy(() => import('./AccountPage' /* webpackChunkName: "AccountPage" */));
const InterviewPage = lazy(() => import('./InterviewPage' /* webpackChunkName: "InterviewPage" */));
const QuestionsPage = lazy(() => import('./QuestionsPage' /* webpackChunkName: "Questions" */));
const CommentsAuthorPage = lazy(() =>
  import('./Comment/CommentsAuthorPage' /* webpackChunkName: "CommentsAuthor" */),
);
const ConfirmationPage = lazy(() =>
  import('./ConfirmationPage' /* webpackChunkName: "Confirmation" */),
);
const FaqPage = lazy(() => import('./FaqPage' /* webpackChunkName: "FAQ" */));

const App = () => (
  <Fragment>
    <Helmet>
      <title>Frontview: Interview for Everyone</title>
    </Helmet>
    <Header />
    <main id="main">
      <Switch>
        <Route exact path="/" component={HomePage} />
        <Route path="/login" component={isLoggedIn(Waiting(LoginPage))} />
        <Route path="/signup" component={isLoggedIn(Waiting(SignupPage))} />
        <Route path="/me" component={User(Waiting(AccountPage))} />
        <Route path="/questions" component={Waiting(QuestionsPage)} />
        <Route path="/interview" component={User(Waiting(InterviewPage))} />
        <Route path="/comments/:username" component={User(Waiting(CommentsAuthorPage))} />
        <Route path="/confirmation/:token" component={Waiting(ConfirmationPage)} />
        <Route path="/faq" component={Waiting(FaqPage)} />
        <Route component={NotFound} />
      </Switch>
    </main>
    <Footer />
    <FlashList />
    <FaqButton />
  </Fragment>
);

export default App;
