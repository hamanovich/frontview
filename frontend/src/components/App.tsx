import React, { Fragment, lazy } from 'react';
import { Switch, Route } from 'react-router-dom';
import { Helmet } from 'react-helmet';
import ScrollToTop from 'react-scroll-up';
import FontAwesome from 'react-fontawesome';

import Button from 'react-bootstrap/Button';

import HomePage from './HomePage';
import NotFound from './NotFound';
import Header from './layout/Header';
import Footer from './layout/Footer';
import FlashList from './flash/FlashList';
import FaqButton from './shared/FaqButton';

import { isLoggedIn, Customer, Waiting } from '../utils/helpers';
import ScrollToTopRoute from '../utils/ScrollToTopRoute';

const LoginPage = lazy(() =>
  import('./LoginPage' /* webpackChunkName: "Login" */),
);
const SignupPage = lazy(() =>
  import('./SignupPage' /* webpackChunkName: "Signup" */),
);
const AccountPage = lazy(() =>
  import('./AccountPage' /* webpackChunkName: "AccountPage" */),
);
const AboutPage = lazy(() =>
  import('./AboutPage' /* webpackChunkName: "AboutPage" */),
);
const InterviewPage = lazy(() =>
  import('./InterviewPage' /* webpackChunkName: "InterviewPage" */),
);
const QuestionsPage = lazy(() =>
  import('./QuestionsPage' /* webpackChunkName: "Questions" */),
);
const CommentsAuthorPage = lazy(() =>
  import('./Comment/CommentsAuthorPage' /* webpackChunkName: "CommentsAuthor" */),
);
const ConfirmationPage = lazy(() =>
  import('./ConfirmationPage' /* webpackChunkName: "Confirmation" */),
);

const App = () => (
  <Fragment>
    <Helmet>
      <title>Frontview: Interview for Everyone</title>
    </Helmet>
    <Header />
    <main id="main">
      <ScrollToTopRoute />
      <Switch>
        <Route exact path="/" component={HomePage} />
        <Route path="/login" component={isLoggedIn(Waiting(LoginPage))} />
        <Route path="/signup" component={isLoggedIn(Waiting(SignupPage))} />
        <Route path="/me" component={Customer(Waiting(AccountPage))} />
        <Route path="/questions" component={Waiting(QuestionsPage)} />
        <Route path="/interview" component={Customer(Waiting(InterviewPage))} />
        <Route
          path="/comments/:username"
          component={Customer(Waiting(CommentsAuthorPage))}
        />
        <Route
          path="/confirmation/:token"
          component={Waiting(ConfirmationPage)}
        />
        <Route path="/about" component={Waiting(AboutPage)} />
        <Route component={NotFound} />
      </Switch>
    </main>
    <Footer />
    <FlashList />
    <FaqButton />
    <ScrollToTop showUnder={160} style={{ bottom: 16, right: 16 }}>
      <Button variant="warning">
        <FontAwesome name="chevron-up" />
      </Button>
    </ScrollToTop>
  </Fragment>
);

export default App;
