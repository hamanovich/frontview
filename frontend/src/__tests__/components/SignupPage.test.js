import React from 'react';
import { shallow } from 'enzyme';

import { SignupPage } from '../../components/SignupPage';

describe('<SignupPage/>', () => {
  const props = {
    signup: jest.fn(),
    addFlashMessage: jest.fn(),
    isUserExists: jest.fn(),
  };

  it('renders <SignupPage /> component', () => {
    const component = shallow(<SignupPage {...props} />);

    expect(component).toMatchSnapshot();
  });
});
