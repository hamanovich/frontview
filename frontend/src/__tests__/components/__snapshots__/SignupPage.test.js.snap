// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignupPage/> renders <SignupPage /> component 1`] = `
<ForwardRef(Bootstrap(Container))>
  <ForwardRef(Bootstrap(Row))>
    <HelmetWrapper
      defer={true}
      encodeSpecialCharacters={true}
    >
      <title>
        Frontview: Register. Welcome
      </title>
    </HelmetWrapper>
    <ForwardRef(Bootstrap(Col))
      md={
        Object {
          "offset": 2,
          "span": 8,
        }
      }
      xl={
        Object {
          "offset": 3,
          "span": 6,
        }
      }
    >
      <h1>
        <FontAwesome
          name="user-plus"
        />
         Register. Welcome
      </h1>
      <ReduxForm
        addFlashMessage={[MockFunction]}
        isUserExists={[MockFunction]}
        signup={[MockFunction]}
      />
    </ForwardRef(Bootstrap(Col))>
  </ForwardRef(Bootstrap(Row))>
</ForwardRef(Bootstrap(Container))>
`;
